sql/core/src/main/scala/org/apache/spark/sql/execution/metric/SQLMetrics.scala

```diff
@@ -57,6 +57,11 @@ class SQLMetric(val metricType: String, initValue: Long = 0L) extends Accumulato
 
   override def add(v: Long): Unit = _value += v
 
+  // Because `SQLMetric` only stores long value, in order to store double average metrics, we
+  // multiply the given double with a base integer. When showing the metrics, it will be
+  // divided by the base integer to restore back a double.
+  def setWithDouble(v: Double): Unit = _value = (v * SQLMetrics.baseForAvgMetric).toLong
```

Comment:
To store a float value into `SQLMetric`, currently I have no better idea to do. Any suggestion welcome.